### DISCUSSION TITLE
How does the Neuron client determine whether the CKB node needs to be upgraded?

### DISCUSSION THREAD

**Discussion by eval-exec** [nervosnetwork/neuron#3202] [Q&A] [answered]
When I started the Neuron client, I noticed the following log:
```
[2024-07-05T08:05:17.300Z] [info]  Update for version 0.116.2 is not available (latest version: 0.116.2, downgrade is disallowed).
```

Does this mean that whenever nervosnetwork/ckb officially releases a new version, the Neuron client will automatically upgrade the CKB node version?

**Comment by Keith-CY** [comment, marked as answer]
> When I started the Neuron client, I noticed the following log:
> 
> ```
> [2024-07-05T08:05:17.300Z] [info]  Update for version 0.116.2 is not available (latest version: 0.116.2, downgrade is disallowed).
> ```
> 
> Does this mean that whenever nervosnetwork/ckb officially releases a new version, the Neuron client will automatically upgrade the CKB node version?

Hi @eval-exec 

The log

```log
[2024-07-05T08:05:17.300Z] [info]  Update for version 0.116.2 is not available (latest version: 0.116.2, downgrade is disallowed).
```

is about checking updates of Neuron app by comparing `latest.yml` from https://github.com/nervosnetwork/neuron/releases/latest

The check will be performed every time the Neuron launches and a `new` badge will be displayed if there's a new release. Also, the `update` button in `Neuron => Settings => General => Version` will be available.

---

The built-in CKB node is bundled when the app is released and cannot be upgraded independently.

The versions of built-in CKB Node and Light Client are set in the code
CKB Node: https://github.com/nervosnetwork/neuron/blob/master/.ckb-version
Light Client: https://github.com/nervosnetwork/neuron/blob/master/.ckb-light-version

Connecting to an external CKB Node is allowed, so users can connect to the latest CKB Node if a new Neuron is not published. The compatibility table is available at https://neuron.magickbase.com/download
